src/test/java/com/aws/greengrass/authorization/WildcardTrieTest.java

```diff
@@ -237,25 +297,31 @@ void testMultipleWildcardsMatching() {
         WildcardTrie rt = new WildcardTrie();
         rt.add("xyz*/+/*room/#");
         assertTrue(rt.matchesMQTT("xyzzzz/89/bedroom/light/5"));
-        assertFalse(rt.matchesMQTT("xyzzzz/89/fan2/bedroom/light/5"));
+        assertTrue(rt.matchesMQTT("xyzzzz/89/fan2/bedroom/light/5"));
 
         assertFalse(rt.matchesStandard("xyzzzz/89/bedroom/light/5"));
         assertFalse(rt.matchesStandard("xyzzzz/89/fan2/bedroom/light/5"));
         assertTrue(rt.matchesStandard("xyzzzz/+/bedroom/#"));
 
         rt.add("12/*#");
         assertTrue(rt.matchesMQTT("12/345#"));
+        assertTrue(rt.matchesMQTT("12/34/5#"));
```

Comment:
These aren't valid topic filters according to the spec. We probably shouldn't even be accepting them at all

http://docs.oasis-open.org/mqtt/mqtt/v3.1.1/os/mqtt-v3.1.1-os.html#_Toc398718106

Reply:
We don't check that here as it's a general resource, but that check is implemented in Pubsub and MQTT IPC code when operation is called for that particular resource.

Reply:
Had a chat offline about this.

In general I am very uncomfortable with this code allowing actions on invalid resources. This code is aware of MQTT semantics. `matchesMQTT` should NEVER return true on invalid MQTT.

That said, fixing this will require some larger refactoring which can be done later. This method is being supported for legacy reasons, and we want to ensure that nothing else uses the MQTT style policy moving forward. Rather than give callers the option to specify the policy type, we are instead going to force this to happen during trie creation. Only MQTT proxy operations will use this policy type, and it will be hard coded inside the auth module itself rather than being provided by the caller as part of the `isAuthorized` call.